spec/unit/outputs/elasticsearch_spec.rb

```diff
@@ -271,6 +271,45 @@
       end
     end
 
+    describe "with event integration metadata" do
+      context "when there isn't any index setting specified and the event contains an integration metadata index" do
```

Comment:
From line 192, which appears to be the outer context group for this example context, the `options` includes an `index` directive.

When an `index` is explicitly given in the plugin configuration, it _must_ supercede the metadata.

As-specified, explicit configuration _always_ takes precedence over implicit discovery.